device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsSessionHandler.java

```diff
@@ -35,9 +35,6 @@ class AmqpsSessionHandler extends BaseHandler implements AmqpsLinkStateCallback
     private IotHubTransportMessage explicitInProgressTwinSubscriptionMessage;
     private IotHubTransportMessage explicitInProgressMethodsSubscriptionMessage;
 
-    //Carries over state between reconnections
-    private boolean subscribeToMethodsOnReconnection = false;
-    private boolean subscribeToTwinOnReconnection = false;
```

Comment:
Keep in mind that not every user wants to subscribe to methods/twin. These flags are in place to track if the session was once subscribed to methods/twin

Reply:
reverted changes to AmqpsSessionhandler and making sure after closing amqp connection due to transport error, we save previous AmqpsSessions so that their twin/methods subscriptions can persist.